packages/connect-node/src/connect-protocol.ts

```diff
@@ -519,6 +570,22 @@ export function createConnectProtocol(
             const type = connectParseContentType(
               requestHeader.get(headerContentType)
             );
+            const timeout = parseInt(
+              requestHeader.get(connectTimeoutHeader) ?? ""
+            );
```

Comment:
I don't think we can get away with `parseInt` here (it also accepts exponential notation and negative values). Best to add a connectParseTimeoutMs() function to connect-core. 

If parsing of the timeout value failed, you need to call `endWithConnectEndStream()` for all streaming methods, and `endWithConnectUnaryError()` for unary methods, and stop processing. 

Please parse the timeout _after_ checking whether the content-type is valid. We should not try to parse the `Connect-Timeout-Ms` header if we don't recognize the `content-type` header. In case of the Connect protocol, we should also fail on the `Connect-Protocol-Version` header before we fail on an invalid timeout.